client/context.go

```diff
@@ -218,12 +220,26 @@ func (ctx Context) WithInterfaceRegistry(interfaceRegistry codectypes.InterfaceR
 // WithViper returns the context with Viper field
 func (ctx Context) WithViper() Context {
 	v := viper.New()
-	// Automatically overrides config field by env variable if this exists
-	v.AutomaticEnv()
 	ctx.Viper = v
 	return ctx
 }
 
+// WithHomeFlag checks if home flag is changed.
+// If this is a case, we update HomeDir field of Client Context
+/* Discovered a bug with Cory
+./build/simd init andrei --home ./test
+cd test/config there is no client.toml configuration file
+*/
+
+func (ctx Context) WithHomeFlag(cmd *cobra.Command) Context {
```

Comment:
There is a bug by executing this command `./build/simd init andrei --home ./test`. There is no `client.toml` created inside ./test/config. WithHomeFlag fixes this bug.